modules/oagw/oagw-sdk/src/api.rs

```diff
@@ -0,0 +1,189 @@
+        &self,
+        ctx: &SecurityContext,
+        req: OagwInvokeRequest,
+    ) -> Result<OagwInvokeResponse, OagwError>;
+
+    /// Invoke an outbound API with streaming response (e.g., SSE).
+    ///
+    /// # Arguments
+    /// * `ctx` - Security context for authorization
+    /// * `req` - Invocation request with route, method, path, etc.
+    ///
+    /// # Returns
+    /// * `Ok(OagwResponseStream)` - Stream of response chunks
+    /// * `Err(OagwError)` - Error if stream cannot be established
+    ///
+    /// # Note
+    /// Streaming requests are never retried by OAGW. If the stream fails mid-way,
+    /// it yields a terminal `OagwStreamAbort` with resume hints if applicable.
+    async fn invoke_stream(
+        &self,
+        ctx: &SecurityContext,
+        req: OagwInvokeRequest,
+    ) -> Result<OagwResponseStream, OagwError>;
+
+    // === Route CRUD ===
+
+    /// Create a new outbound API route.
+    async fn create_route(
+        &self,
+        ctx: &SecurityContext,
+        new_route: NewRoute,
+    ) -> Result<Route, OagwError>;
+
+    /// Get a route by ID.
+    async fn get_route(&self, ctx: &SecurityContext, id: Uuid) -> Result<Route, OagwError>;
+
+    /// List routes with `OData` pagination.
+    async fn list_routes(
+        &self,
+        ctx: &SecurityContext,
+        query: ODataQuery,
+    ) -> Result<Page<Route>, OagwError>;
+
+    /// Update a route with partial data.
+    async fn update_route(
+        &self,
+        ctx: &SecurityContext,
+        id: Uuid,
+        patch: RoutePatch,
+    ) -> Result<Route, OagwError>;
+
+    /// Delete a route by ID.
+    async fn delete_route(&self, ctx: &SecurityContext, id: Uuid) -> Result<(), OagwError>;
+
+    // === Link CRUD ===
+
+    /// Create a new outbound API link.
+    async fn create_link(
+        &self,
+        ctx: &SecurityContext,
+        new_link: NewLink,
+    ) -> Result<Link, OagwError>;
+
+    /// Get a link by ID.
+    async fn get_link(&self, ctx: &SecurityContext, id: Uuid) -> Result<Link, OagwError>;
+
+    /// List links with `OData` pagination.
+    async fn list_links(
+        &self,
+        ctx: &SecurityContext,
+        query: ODataQuery,
+    ) -> Result<Page<Link>, OagwError>;
+
+    /// Update a link with partial data.
+    async fn update_link(
+        &self,
+        ctx: &SecurityContext,
+        id: Uuid,
+        patch: LinkPatch,
+    ) -> Result<Link, OagwError>;
+
+    /// Delete a link by ID.
+    async fn delete_link(&self, ctx: &SecurityContext, id: Uuid) -> Result<(), OagwError>;
+}
+
+/// Plugin API trait implemented by OAGW plugins.
+///
+/// Plugins register themselves with a scoped client in `ClientHub`:
+/// ```ignore
+/// let scope = ClientScope::gts_id(&instance_id);
+/// hub.register_scoped::<dyn OagwPluginApi>(scope, plugin_impl);
+/// ```
+///
+/// The gateway resolves plugins at runtime based on protocol and auth type requirements.
+#[async_trait]
+pub trait OagwPluginApi: Send + Sync {
```

Comment:
I suggest to use XXXClient suffix. It's really client for some API but not API itself.